src/features/DisplayControls/Allergies/components/Allergies.jsx

```diff
@@ -0,0 +1,169 @@
+import {
+  DataTable,
+  DataTableSkeleton,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "carbon-components-react";
+import React, { useEffect, useState } from "react";
+import { useFetchAllergiesIntolerance } from "../hooks/useFetchAllergiesIntolerance";
+import PropTypes from "prop-types";
+import "../styles/Allergies.scss";
+import { FormattedMessage } from "react-intl";
+
+const Allergies = (props) => {
+  const { patientId } = props;
+
+  const { allergiesData, isLoading } = useFetchAllergiesIntolerance(patientId);
+  const [rows, setRows] = useState([]);
+  const NoAllergenMessage = (
+    <FormattedMessage
+      id={"NO_ALLERGENS_MESSAGE"}
+      defaultMessage={"No Allergen is captured for this patient yet."}
+    />
+  );
+
+  useEffect(() => {
+    if (allergiesData && allergiesData.entry) {
+      const allergies = [];
+      allergiesData.entry?.map((allergy) => {
+        allergies.push({
```

Comment:
we can directly assign the return object to allergies variable and no need to use push in map, just use return instead of push and assign directly to allergies variable